src/pudl/analysis/plant_parts_eia.py

```diff
@@ -1558,6 +1565,8 @@ def match_to_single_plant_part(
             validate="m:m",
             suffixes=("_og", ""),
         )
+        # there should be no records without a matching generator
+        assert ~(part_df.record_id_eia.isnull().values.any())
```

Comment:
Hmmm maybe I use `pudl.validate.no_null_rows()`

Reply:
Conceptually I agree, but because we run the full validations so infrequently.... i think it makes sense to do this assert here. perhaps when we move to the full etl + validations in our nightly builds moving this over there would be a good idea.